signature/signature.go

```diff
@@ -27,37 +27,70 @@ func (err InvalidSignatureError) Error() string {
 }
 
 // Signature is a parsed content of a signature.
+// The only way to get this structure from a blob should be as a return value from a successful call to verifyAndExtractSignature below.
 type Signature struct {
 	DockerManifestDigest digest.Digest
 	DockerReference      string // FIXME: more precise type?
 }
 
-// Wrap signature to add to it some methods which we don't want to make public.
-type privateSignature struct {
-	Signature
+// untrustedSignature is a parsed content of a signature.
+type untrustedSignature struct {
+	UntrustedDockerManifestDigest digest.Digest
+	UntrustedDockerReference      string // FIXME: more precise type?
+	UntrustedCreatorID            *string
+	UntrustedTimestamp            *int64
```

Comment:
Actually, there is no risk/tradeoff here; we can do the `UnmarshalJSON` into an `int64` in the private `untrustedSignature`, and do the conversion to `time.Time` in `GetUntrustedSignatureInformationWithoutVerifying`. Changed like that.